src/xdocs/config_imports.xml

```diff
@@ -735,25 +735,107 @@ import android.*;
+          access <code>dao</code> and <code>service</code> layer from <code>ui</code> packages. The
+          root package is also a regular expression that is used to handle old and new domain name
+          with the same rules.
+        </p>
+
+        <source>
+&lt;import-control pkg=&quot;(de.olddomain|de.newdomain)\..*&quot; regex=&quot;true&quot;&gt;
+    &lt;subpackage pkg=&quot;[^.]+\.dao&quot; regex=&quot;true&quot;&gt;
+        &lt;disallow pkg=&quot;.*\.ui&quot; regex=&quot;true&quot;/&gt;
+        &lt;disallow pkg=&quot;.*\.(awt|swing).\.*&quot; regex=&quot;true&quot;/&gt;
+    &lt;/subpackage&gt;
+    &lt;subpackage pkg=&quot;[^.]+\.ui&quot; regex=&quot;true&quot;&gt;
+        &lt;disallow pkg=&quot;.*\.(dao|service)&quot; regex=&quot;true&quot;/&gt;
+    &lt;/subpackage&gt;
+&lt;/import-control&gt;
+        </source>
+
+        <p>
+          For a real-life import control file look at the file called <a
           href="https://github.com/checkstyle/checkstyle/blob/master/config/import-control.xml">
           import-control.xml</a>
           which is part of the Checkstyle distribution.
         </p>
+
+        <h4 id="regex-notes">Notes on regular expressions</h4>
+        <p>
+          Regular expressions in import rules have to match either Java packages or
+          classes. The language rules for packages and class names can be described by the
+          following complicated regular expression that takes into account that Java names may
+          contain any unicode letter, numbers, underscores, and dollar signs (see section 3.8
+          in the <a href="http://docs.oracle.com/javase/specs/">Java specs</a>):
+        </p>
+        <ul>
+          <li>
+            <code>[\p{Letter}_$][\p{Letter}\p{Number}_$]*</code> or short
+            <code>[\p{L}_$][\p{L}\p{N}_$]*</code> for a class name or package component.
+          </li>
+          <li>
+            <code>([\p{L}_$][\p{L}\p{N}_$]*\.)*[\p{L}_$][\p{L}\p{N}_$]*</code>
+            for a fully qualified name.
+          </li>
+        </ul>
+        <p>
+          But it is not necessary to use these complicated expressions since no validation is
+          required. Differentiating between package separator '.' and others is
+          sufficient. Unfortunately '.' has a special meaning in regular expressions so one has
+          to write <code>\.</code> to match an actual dot.
+        </p>
+        <ul>
+          <li>
+            Use <code>[^.]+</code> (one or more "not a dot" characters) for a class name or
+            package component.
+          </li>
+          <li>
+            Use <code>com\.google\.common\.[^.]+</code> to match any subpackage of
+            <code>com.google.common</code>.
+          </li>
+          <li>
+            When matching concrete packages like <code>com.google.common</code> omitting the
+            backslash before the dots may improve readability and may be just exact enough:
```

Comment:
Great. Please keep in mind that this is a section with advices which grew out of my own usage of regular expression in this context and out of attempts to help others.